From 36c4b00496c27b90bb3b0b9d26ba0ce23e5c7a71 Mon Sep 17 00:00:00 2001
From: Danny Nold <dannynold@freescale.com>
Date: Thu, 31 Mar 2011 22:23:26 -0500
Subject: [PATCH] ENGR00140737-1 - ARM: plat-mxc: Disable child clocks before parent clocks

- Reversed ordering in which clocks are disabled.  Child clocks should
be disabled before parent (root) clocks.

Signed-off-by: Danny Nold <dannynold@freescale.com>
---
 arch/arm/plat-mxc/clock.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/plat-mxc/clock.c b/arch/arm/plat-mxc/clock.c
index 8551456..34c4be1 100644
--- a/arch/arm/plat-mxc/clock.c
+++ b/arch/arm/plat-mxc/clock.c
@@ -4,7 +4,7 @@
  * Copyright (C) 2004 - 2005 Nokia corporation
  * Written by Tuukka Tikkanen <tuukka.tikkanen@elektrobit.com>
  * Modified for omap shared clock framework by Tony Lindgren <tony@atomide.com>
- * Copyright 2007-2010 Freescale Semiconductor, Inc.
+ * Copyright 2007-2011 Freescale Semiconductor, Inc.
  * Copyright 2008 Juergen Beisert, kernel@pengutronix.de
  *
  * This program is free software; you can redistribute it and/or
@@ -141,11 +141,11 @@ static void __clk_disable(struct clk *clk)
 		return;
 
 	if (!(--clk->usecount)) {
-		__clk_disable(clk->parent);
-		__clk_disable(clk->secondary);
-
 		if (clk->disable)
 			clk->disable(clk);
+
+		__clk_disable(clk->secondary);
+		__clk_disable(clk->parent);
 	}
 }
 
-- 
1.5.4.4

